### PR TITLE
doc: extensions: Add shield support to zephyr-app-commands

### PR DESCRIPTION
**Pull request by galak**
Add :shield: as an argument to zephyr-app-commands, update the shield
porting rst example to use it.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: doc/extensions/zephyr/application.py [hunk 1]
```diff
@@ -48,6 +48,9 @@ class ZephyrAppCommandsDirective(Directive):
     \:board:
       if set, the application build will target the given board.
 
+    \:shield:
```
Comment (dbkinder) [inline doc/extensions/zephyr/application.py]: Please update the documentation in ``doc/documentation/doc-guidelines.rst`` with this new argument too.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/extensions/zephyr/application.py (hunk 1 above)
Comment (dbkinder) [reply doc/extensions/zephyr/application.py]: ... actually I'm making edits to the doc-guidelines.rst file right now, so I'll just add it while I'm there.

**Review by dbkinder — APPROVED** [review]
I'll make the addition to the doc guidelines

**Comment by nashif** [comment]
> Add 🛡 as an argument...

wow, a real shield

**Pull request merged**